tidy3d/components/monitor.py

```diff
@@ -512,20 +575,17 @@ class ModeSolverMonitor(AbstractModeMonitor):
         "dimension.",
     )
 
+    colocate: bool = pydantic.Field(
```

Comment:
just curious: to override the `colocate` field in the parent class, why not use a validator to hard code it, instead of making a new field in the subclass?

Reply:
I think that way is also possible, I didn't really consider it. One advantage of the current implementation is that it should be directly readable from the docs what the allowed values are.

On the other hand, that does make me wonder: do we really want `colocate` to appear as a Field in all the monitors in which it cannot be modified? My initial approach was to have a property e.g. `colocate_in_solver` that is set for the monitors for which we set it internally, and set by `self.colocate` for monitors where we allow a `colocate` Field. This essentially introduces an extra attribute, but show `colocate` as an input argument to monitors for which it doesn't apply. Same goes for `interval_space` and say `record_in_pml` in the future. @tylerflex suggested the `Literal` approach instead, but does that make the api more confusing?

Reply:
The `Literal` approach does seem most similar to what one might see in a GUI, where the `colocate` field can be "greyed out" and not modifiable for some monitors, but essentially a drop-down menu for others?